src/test/java/com/team701/buddymatcher/controllers/communication/CommunicationControllerIntegrationTest.java

```diff
@@ -94,6 +95,42 @@ void getChatList() throws Exception {
     }
 
     @Test
+    void deleteMessages() throws Exception {
+        // Test that the messages exist before deletion
+        mvc.perform(get("/api/communication/messages/{id}",2)
+                        .sessionAttrs(Collections.singletonMap("UserId", 1)))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0].id").value(1))
+                .andExpect(jsonPath("$[0].senderId").value(1))
+                .andExpect(jsonPath("$[0].receiverId").value(2))
+                .andExpect(jsonPath("$[0].timestamp").value("1647394176000"))
+                .andExpect(jsonPath("$[0].content").value("HARRO"))
+                .andExpect(jsonPath("$[0].read").value(true))
+                .andExpect(jsonPath("$[1].id").value(2))
+                .andExpect(jsonPath("$[1].senderId").value(2))
+                .andExpect(jsonPath("$[1].receiverId").value(1))
+                .andExpect(jsonPath("$[1].content").value("Yo HARRO"))
+                .andExpect(jsonPath("$[1].timestamp").value("1647394821000"))
+                .andExpect(jsonPath("$[1].read").value(false))
+                .andDo(print());
+
+        // Test deleting all messages
+        mvc.perform(delete("/api/communication/messages/{id}",2)
+                        .sessionAttrs(Collections.singletonMap("UserId", 1)))
+                .andExpect(status().isOk());
+
+        // Test that all the messages have been deleted successfully
+        mvc.perform(get("/api/communication/messages/{id}",2)
+                        .sessionAttrs(Collections.singletonMap("UserId", 1)))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$").isEmpty());
+
+        // Test when the input user is not found
+        mvc.perform(delete("/api/communication/messages/{id}",4)
+                        .sessionAttrs(Collections.singletonMap("UserId", 1)))
+                .andExpect(status().isNotFound());
```

Comment:
This is failing: _Status expected:<404> but was:<200>_

Reply:
This is passing now. It was not throwing an error when the two users weren't buddies (which means they have a chat), but now it checks for it and throws an exception accordingly.